tests/apps/test_data.py

```diff
@@ -328,3 +332,111 @@ def test_data_dim_correct(self, dataset_patched):
         assert m == dataset_patched.n_features
         assert p == len(dataset_patched.adjs)
         assert q == len(dataset_patched.unit_data)
+
+
+@pytest.mark.parametrize("datasets", WATER_DATASETS_LIST)
+class TestWaterDatasets:

```

Comment:
It seems like these tests look general but actually are quite specific to the water dataset. Maybe best to just specialise to water for now.